src/alchemlyb/tests/test_preprocessing.py

```diff
@@ -213,6 +236,97 @@ def test_raise_ValueError_for_mismatched_data(self, series):
         with pytest.raises(ValueError):
             self.slicer(data, series=series)
 
+    @pytest.mark.parametrize(('data', 'lower', 'upper'),
+                             [
+                                 (gmx_benzene_dHdl(), 1000, 34000),
+                                 (gmx_benzene_u_nk(), 1000, 34000),
+                             ])
+    def test_lower_and_upper_bound_slicer(self, data, lower, upper):
+        """
+        Test that the lower and upper time is respected when using statistical_inefficiency
+        without a series. In this case, statistical_inefficiency should behave like slicing
+        """
+        original_length = len(data)
+        # Check that the input data is appropriate for the test
+        assert any(data.reset_index()['time'] < lower)
+        assert any(data.reset_index()['time'] > upper)
+
+        # Slice data, and check that we don't observe times outside
+        # the prescribed range
+        sliced = self.slicer(data,
+                             series=None,
```

Comment:
Sorry, I didn't quite see how is this different from `test_lower_and_upper_bound` as the series is default to None?

Reply:
Well, this test class tests `statistical_inefficiency(...)`, the one above tests `slicing(...)`.

So this tests makes sure that `statistical_inefficiency` respects the upper and lower bounds when it's used without series. The test above makes sure that `slicing` respects the upper and lower bounds.

This test could be rewritten as equivalence between `statistical_inefficiency` and `slicing`, if you prefer.

Reply:
@ptmerz  Why do you need to check slicing when you are changing the code in statistical_inefficiency?

Reply:
I don't _need_ to. While investigating the bug I reported I noticed that there was no test for ranges in `slicing` or in `statistical_inefficiency`, which made it possible for the bug in `statistical_inefficiency` to go unnoticed until now. I thought it would be helpful to broaden the test coverage also for `slicing` to avoid any future bugs. Do you prefer me to remove them?